jax/random.py

```diff
@@ -261,8 +291,8 @@ def fold_in(key, data):
 
 @jit
 def _fold_in(key, data):
-  key2 = lax.tie_in(key, PRNGKey(data))
-  return threefry_2x32(key, key2)
+  key2 = lax.tie_in(key, np.asarray([[0, data]], dtype='uint32'))
+  return threefry_2x32(key, key2)[0]
```

Comment:
I'm surprised at this indexing; it seems like the old version returned the full output of `threefry_2x32`. What changed? Did this change the type of the function?

Reply:
This is just because threefry now returns a shape (1, 2) array where before it returned shape (2,). I could use a reshape/ravel or something to get the same effect here.

Reply:
Maybe getting rid of vamp (re-enabling rank polymorphism) would be a better solution 🙂